python/katana/dynamic_bitset.pyx

```diff
@@ -0,0 +1,67 @@
+from .cpp.libgalois.DynamicBitset cimport DynamicBitset as CDynamicBitset
+
+
+cdef class DynamicBitset:
+    cdef CDynamicBitset underlying
+
+    def __cinit__(self):
+        self.underlying = CDynamicBitset()
+
+    def resize(self, n):
+        """Resizes the bitset."""
+        self.underlying.resize(n)
+
+    def clear(self):
+        """Resets all bits and sets the size of the set to 0."""
+        self.underlying.clear()
+
+    def shrink_to_fit(self):
+        """Shrinks the allocation for bitset to its current size."""
+        self.underlying.shrink_to_fit()
+
+    def __len__(self):
+        """Gets the number of bits held by the bitset."""
+        return self.underlying.size()
+
+    def reset(self):
+        """Unsets every bit in the bitset."""
+        self.underlying.reset()
+
+    def __getitem__(self, index):
+        """Checks a bit to see if it is currently set.
+
+        This is not safe w.r.t. setting bits."""
+        if not isinstance(index, int):
+            raise TypeError('index has to be an integer')
+
+        if index < 0 or index >= len(self):
+            raise IndexError(f'{index} out of range')
```

Comment:
The convention is that `IndexError`s argument is just the index and nothing else (not even converted to a string).